### PR TITLE
Post-callout side effect injection.

### PR DESCRIPTION
**Pull request by andersio**
All events have options of pre- and post-callout injection, except for `value`. This PR adds a `valueSent` injection point to `Signal.on` and `SignalProducer.on`. It also reordered the arguments in `SignalProducer.on` to match `Signal.on`.

**Rationale**
It wasn't possible to inject a side effect **_after_** every value is consumed by a `BindingTarget`.
```
textField.stringValue <~ viewModel.name
    .on(value: { _ in print("This happens before the consumption") })

// You can wrap a binding target in a binding target though.
didSet(textField.stringValue) {
    // My "didSet" side effect.
} <~ viewModel.name
```

### CODE REVIEW

File: Sources/Signal.swift [hunk 1]
```diff
@@ -844,6 +850,9 @@ extension SignalProtocol {
 				switch receivedEvent {
 				case let .value(v):
 					value?(v)
+					observer.send(value: v)
```
Comment (NachoSoto) [inline Sources/Signal.swift]: I'm confused, why is this line added here too? Aren't we sending events twice now?

File: Sources/SignalProducer.swift [hunk 2]
```diff
@@ -1242,32 +1242,37 @@ extension SignalProducerProtocol {
...
 		event: ((Event<Value, Error>) -> Void)? = nil,
-		value: ((Value) -> Void)? = nil,
 		failed: ((Error) -> Void)? = nil,
 		completed: (() -> Void)? = nil,
 		interrupted: (() -> Void)? = nil,
 		terminated: (() -> Void)? = nil,
-		disposed: (() -> Void)? = nil
+		disposed: (() -> Void)? = nil,
+		valueSent: ((Value) -> Void)? = nil,
+		value: ((Value) -> Void)? = nil
```
Comment (NachoSoto) [inline Sources/SignalProducer.swift]: The ordering was on purpose, can you move `value` up again?

File: Sources/Signal.swift (hunk 1 above)
Comment (andersio) [inline Sources/Signal.swift]: There is a `return` below as an early out.

**Comment by andersio** [comment]
@NachoSoto True. Just I almost forgot that `BindingTarget` could be asynchronous.
